src/ra_server.erl

```diff
@@ -6,6 +6,8 @@
 %%
 -module(ra_server).
 
+-feature(maybe_expr, enable).
```

Comment:
Can we refactor all code not to use `maybe` expressions? Currently it has broken both the OTP 24 _and_ 25 tests. We can probably fix the 25 tests but it would be nice to still stay 24 compatible a little longer (as it is a library).

```diff --git a/src/ra.erl b/src/ra.erl
index 159628f..8202a7b 100644
--- a/src/ra.erl
+++ b/src/ra.erl
@@ -415,7 +415,7 @@ start_cluster(System, [#{cluster_name := ClusterName} | _] = ServerConfigs,
                     ok  -> true;
                     Err ->
                         ?ERR("ra: failed to start a server ~w, error: ~p",
-                              [C, Err]),
+                              [maps:get(id, C), Err]),
                         false
                 end
             end, ServerConfigs),
diff --git a/src/ra_server.erl b/src/ra_server.erl
index a16fe40..f4f60a0 100644
--- a/src/ra_server.erl
+++ b/src/ra_server.erl
@@ -6,8 +6,6 @@
 %%
 -module(ra_server).
 
--feature(maybe_expr, enable).
-
 -include("ra.hrl").
 -include("ra_server.hrl").
 
@@ -2947,22 +2945,27 @@ get_membership(#{cfg := #cfg{id = Id, uid = UId}, cluster := Cluster} = State) -
     Default = maps:get(membership, State, voter),
     get_membership(Cluster, Id, UId, Default).
 
--spec maybe_promote_peer(ra_server_id(), ra_server_state(), effects()) -> effects().
+-spec maybe_promote_peer(ra_server_id(), ra_server_state(), effects()) ->
+    effects().
 maybe_promote_peer(PeerId, #{cluster := Cluster}, Effects) ->
-    maybe
+    case Cluster of
         #{PeerId := #{match_index := MI,
-                      voter_status := VoterStatus}} ?= Cluster,
-        #{membership := promotable, target := Target} ?= VoterStatus,
-        true ?= MI >= Target,
-        E = {next_event,
-          {command, {'$ra_join',
-                    #{ts => os:system_time(millisecond)},
-                    #{id => PeerId,
-                      voter_status => VoterStatus#{membership => voter}},
-                    noreply}}},
-        [E | Effects]
-    else
-        _ -> Effects
+                      voter_status := VoterStatus}}  ->
+            case VoterStatus of
+                #{membership := promotable, target := Target} when
+                      MI >= Target ->
+                    [{next_event,
+                      {command, {'$ra_join',
+                                 #{ts => os:system_time(millisecond)},
+                                 #{id => PeerId,
+                                   voter_status => VoterStatus#{membership => voter}},
+                                 noreply}}},
+                     Effects];
+                _ ->
+                    Effects
+            end;
+        _ ->
+            Effects
     end.
 
 -spec required_quorum(ra_cluster()) -> pos_integer().
diff --git a/src/ra_server_proc.erl b/src/ra_server_proc.erl
index 16326ad..c985aa4 100644
--- a/src/ra_server_proc.erl
+++ b/src/ra_server_proc.erl
@@ -7,8 +7,6 @@
 %% @hidden
 -module(ra_server_proc).
 
--feature(maybe_expr, enable).
-
 -behaviour(gen_statem).
 
 -compile({inline, [handle_raft_state/3]}).
diff --git a/test/coordination_SUITE.erl b/test/coordination_SUITE.erl
index 4e8f065..c6a45df 100644
--- a/test/coordination_SUITE.erl
+++ b/test/coordination_SUITE.erl
@@ -56,6 +56,10 @@ groups() ->
 
 init_per_suite(Config) ->
     %% as we're not starting the ra application and we want the logs
+    LogFile = filename:join(?config(priv_dir, Config), "ra.log"),
+    LogCfg = #{config => #{file => LogFile}},
+    logger:add_handler(test_handler, logger_std_h, LogCfg),
+    logger:set_primary_config(level, debug),
     ra_env:configure_logger(logger),
     Config.
 
@@ -112,6 +116,7 @@ start_stop_restart_delete_on_remote(Config) ->
     ok.
 
 start_cluster(Config) ->
+    logger:set_primary_config(level, debug),
     PrivDir = ?config(data_dir, Config),
     ClusterName = ?config(cluster_name, Config),
     NodeIds = [{ClusterName, start_follower(N, PrivDir)} || N <- [s1,s2,s3,s4,s5]],
@@ -722,7 +727,7 @@ start_follower(N, PrivDir) ->
     ct:pal("starting secondary node with ~ts on host ~ts for node ~ts", [Pa, Host, node()]),
     {ok, S} = slave:start_link(Host, N, Pa),
     ok = ct_rpc:call(S, ?MODULE, node_setup, [PrivDir]),
-    ok = erpc:call(S, ra, start, []),
+    ok = ct_rpc:call(S, ra, start, []),
     ok = ct_rpc:call(S, logger, set_primary_config,
                      [level, all]),
     S.
```